rspec-mode.el

```diff
@@ -225,6 +225,13 @@ info, are considered errors."
   :type 'boolean
   :group 'rspec-mode)
 
+(defcustom rspec-holder-dirs '("app" "lib")
```

Comment:
@dgutov, i don't care for `applike`. It is too rails centric. Most non-rails projects have source code in `lib`. We could use `liblike` but it seems just as lame. This whole issue seems even more awkward when i consider the `lib/` directory in a rails app. Specs for files in the `lib` of a rails project would, optimally, go into `spec/lib/`. However, for non-rails projects they should go into `spec/`. 

Is the logic we want really something like: files below the "main" source dir get that dir name trimmed, all others have the same structure below the spec dir as they do below the project root? so 

 * `myapp/app/foo.rb` -> `myapp/spec/foo_spec.rb`
 * `myapp/lib/bar.rb` -> `myapp/spec/lib/bar_spec.rb`
 * `myapp/services/baz.rb` -> `myapp/spec/services/baz_spec.rb`
but
 * `mygem/lib/foo.rb` -> `mygem/spec/foo_spec.rb`

If that is the desired behavior we can avoid the customization variable by assuming the "primary source" directory is `app` if it exists and otherwise is `lib`. All things being equal, fewer customization variables is better.

Can we think of any scenarios in which this approach produces weird results?